Support for Go 1.11 with GO111MODULES=on

It seems that https://github.com/envoyproxy/go-control-plane/pull/179 introduced a hard dependency on Go 1.12. While 1.11 is no longer supported by the core language team, it is still critical for environments like appengine, where it is still supported. I ran into this problem while trying to compile [`grpcurl` with Go 1.11](https://travis-ci.org/github/fullstorydev/grpcurl/jobs/661748053?utm_medium=notification&utm_source=github_status) ([PR](https://github.com/fullstorydev/grpcurl/pull/137)):

```
go test -race ./...
go build golang.org/x/sys/unix: module requires Go 1.12
go build github.com/cncf/udpa/go/udpa/annotations: module requires Go 1.13
go build github.com/envoyproxy/go-control-plane/envoy/annotations: module requires Go 1.12
go build github.com/envoyproxy/go-control-plane/envoy/type: module requires Go 1.12
FAIL	github.com/fullstorydev/grpcurl [build failed]
FAIL	github.com/fullstorydev/grpcurl/cmd/grpcurl [build failed]
Makefile:82: recipe for target 'test' failed
```

There seems to be a general problem with the Go tooling auto-inserting hard dependencies like this when none was intended by the author. Is that the case here? Or is there some feature that the library truly does depend on in Go 1.12+?

CC @kyessenov 
CC @davecheney

--- a/go.mod
+++ b/go.mod
@@ -1,10 +1,10 @@
 module github.com/envoyproxy/go-control-plane
 
-go 1.12
+go 1.11
 
 require (
 	github.com/census-instrumentation/opencensus-proto v0.2.1
-	github.com/cncf/udpa/go v0.0.0-20191209042840-269d4d468f6f
+	github.com/cncf/udpa/go v0.0.0-20200313221541-5f7e5dd04533
 	github.com/envoyproxy/protoc-gen-validate v0.1.0
 	github.com/golang/protobuf v1.3.2
 	github.com/google/go-cmp v0.2.0

